### DISCUSSION TITLE
getting an stack overflow error 3221225725

### DISCUSSION THREAD

**Discussion by Rahul-Reddy-T** [community/community#126271] [Programming Help]
### Body

#include<stdio.h>
#include<stdlib.h>
void bubblesort(int arr[],int n)
{
    int i,j,temp;
    for(i=0;i<=n-2;i++)
    {
        for(j=0;j<=n-i-2;j++)
        {
            if(arr[j]>arr[j+1])
            {
            temp=arr[j];
            arr[j]=arr[j+1];
            arr[j+1]=temp;
            }
        }
    }
}
void main()
{
    int i,n,arr[n];
    printf("enter no. of elements\n");
    scanf("%d",&n);
    printf("entry values\n");
    for(i=0;i<n;i++)
    {
        scanf("%d",arr[i]);
    }
    bubblesort(arr,n);
    printf("array after sorting\n");
    for(i=0;i<n;i++)
    {
        printf("%d\t",arr[i]);
    }
   
}

i am trying to run this code from past 30 min but getting an same error exit code=3221225725 in vs code

### Guidelines

- [X] I have read and understood this category's [guidelines](https://github.com/orgs/community/discussions/51384) before making this post.

**Comment by Himanshu-Khadka** [comment]
It looks like you're encountering a runtime error due to uninitialized variables and incorrect handling of array inputs. Let's break down and fix your code step by step:

1. Declare the array after knowing its size.
2. Pass the correct reference of the array elements to `scanf`.
3. Make sure the main function returns an `int`.

Here's the corrected version of your code:

```c
#include <stdio.h>
#include <stdlib.h>

void bubblesort(int arr[], int n) {
    int i, j, temp;
    for (i = 0; i < n - 1; i++) {
        for (j = 0; j < n - i - 1; j++) {
            if (arr[j] > arr[j + 1]) {
                temp = arr[j];
                arr[j] = arr[j + 1];
                arr[j + 1] = temp;
            }
        }
    }
}

int main() {
    int i, n;
    printf("Enter number of elements: \n");
    scanf("%d", &n);

    int arr[n]; // Declare array after knowing its size

    printf("Enter values: \n");
    for (i = 0; i < n; i++) {
        scanf("%d", &arr[i]); // Use &arr[i] to get the address of the element
    }

    bubblesort(arr, n);

    printf("Array after sorting:\n");
    for (i = 0; i < n; i++) {
        printf("%d\t", arr[i]);
    }

    return 0; // main should return an int
}
```

Explanation of Changes:

1. Array Declaration:
   - Declared the array `arr` after reading the size `n`.

2. Correct `scanf` usage:
   - Use `&arr[i]` to correctly reference the address of each element in the array during input.

3. Return Type of `main`:
   - Changed the return type of `main` to `int` and added `return 0;` at the end of the function.

Running the Corrected Code:

Make sure to compile the corrected code and run it. It should now correctly sort the array without runtime errors. Here's how you can compile and run the code in VS Code:

1. Open Terminal in VS Code.
2. Compile the code using `gcc -o bubblesort bubblesort.c` (assuming your file is named `bubblesort.c`).
3. Run the executable using `./bubblesort`.

This should solve your issue and allow your code to run without the exit code error.

**Reply by Rahul-Reddy-T** [reply]
Thank you sir...it was a great help😊
